Tryouts/Visuals/Windows/WindowsVisualUtils/EmbeddedWebView.cs

```diff
@@ -0,0 +1,166 @@
+{
+    public class EmbeddedWebView : NativeControlHost
+    {
+
+        #region CommunicationsObjects Styled Avalonia Property
+        public IDictionary<string, object> CommunicationsObjects
+        {
+            get { return GetValue(CommunicationsObjectsProperty); }
+            set { SetValue(CommunicationsObjectsProperty, value); }
+        }
+
+        public static readonly StyledProperty<IDictionary<string, object>> CommunicationsObjectsProperty =
+            AvaloniaProperty.Register<EmbeddedWebView, IDictionary<string, object>>
+            (
+                nameof(CommunicationsObjects)
+            );
+        #endregion CommunicationsObjects Styled Avalonia Property
+
+
+        #region IsReady Direct Avalonia Property
+        private bool _IsReady = false;
+
+        public static readonly DirectProperty<EmbeddedWebView, bool> IsReadyProperty =
+            AvaloniaProperty.RegisterDirect<EmbeddedWebView, bool>
+            (
+                nameof(IsReady),
+                o => o.IsReady,
+                (o, v) => o.IsReady = v
+            );
+
+        public bool IsReady
+        {
+            get => _IsReady;
+            private set
+            {
+                SetAndRaise(IsReadyProperty, ref _IsReady, value);
+            }
+        }
+        #endregion IsReady Direct Avalonia Property
+
+
+        public WebView2 WebView => _webView;
+
+        WebView2 _webView;
+        public EmbeddedWebView()
+        {
+            _webView = new WebView2();
+
+            _ = Init();
+
+            this.GetObservable(CommunicationsObjectsProperty).Subscribe(OnCommunicationsObjectsSet);
+            this.GetObservable(IsReadyProperty).Subscribe(OnIsReadyChanged);
+        }
+
+        public async Task Init()
+        {
+            await _webView.EnsureCoreWebView2Async(null);
+
+            IsReady = true;
+
+            _webView.CoreWebView2.DOMContentLoaded += CoreWebView2_DOMContentLoaded;
+            _webView.CoreWebView2.ContentLoading += CoreWebView2_ContentLoading;
+        }
+
+
+        private void CoreWebView2_ContentLoading(object? sender, CoreWebView2ContentLoadingEventArgs e)
+        {
+        }
+
+        private bool _domContentLoaded = false;
+        private bool _objectsAdded = false;
+        private void CoreWebView2_DOMContentLoaded(object? sender, CoreWebView2DOMContentLoadedEventArgs e)
+        {
+            _domContentLoaded = true;
+            CallJSInit();
+        }
+
+        void CallJSInit()
+        {
+            if ((!_domContentLoaded) || (!_objectsAdded))
+            {
+                return;
+            }
+
+            _webView.CoreWebView2.ExecuteScriptAsync("window.IsWebView2Ready = true;");
+            _webView.CoreWebView2.ExecuteScriptAsync("console.log('Initialized IsWebView2ready');");
+
+            string js =
+                @"if (window.OnWebView2Initialized){ window.OnWebView2Initialized(); }";
```

Comment:
As far as I see this function is defined in one of the hosted web applications. The embedded web view control should not be aware of what is it hosting.

Reply:
I assume it will be a reserved name to use for doing browser initialization from C# code.